src/Groups/MatGrp.jl

```diff
@@ -0,0 +1,808 @@
+    symplectic_group,
+    unitary_group,
+    GL, GO, GU, SL, SO, Sp, SU
+
+
+
+abstract type AbstractMatrixGroupElem <: GAPGroupElem{GAPGroup} end
+
+"""
+    MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+Type of groups `G` of `n x n` matrices over the ring `R`, where `n = degree(G)` and `R = base_ring(G)`.
+
+At the moment, only rings of type `FqNmodFiniteField` are supported.
+"""
+mutable struct MatrixGroup{RE<:RingElem, T<:MatElem{RE}} <: GAPGroup
+   X::GapObj
+   gens::Vector{<:AbstractMatrixGroupElem}
+   ring::Ring
+   deg::Int
+   descr::Symbol                       # e.g. GL
+   ring_iso::GenRingIso
+   mat_iso::GenMatIso
+
+   function MatrixGroup(m::Int64, F::S) where S<:Ring
+      K = elem_type(S)
+      r = new{K, MatElem{K}}()
+      r.deg = m
+      r.ring = F
+      return r
+   end
+
+   function MatrixGroup(m::Int64, F::S, G_gap::GapObj) where S<:Ring
+      K = elem_type(S)
+      r = new{K, MatElem{K}}()
+      r.deg = m
+      r.ring = F
+      r.X = G_gap
+      return r
+   end
+
+   function MatrixGroup(m::Int64, F::S, L::Vector) where S<:Ring
+      K = elem_type(S)
+      r = new{K, MatElem{K}}()
+      r.deg = m
+      r.ring = F
+      r.gens = L
+      return r
+   end
+end
+
+
+"""
+    MatrixGroupElem{RE<:RingElem, T<:MatElem{RE}} <: GAPGroupElem{MatrixGroup}
+Elements of a group of type `MatrixGroup{RE<:RingElem, T<:MatElem{RE}}`
+"""
+mutable struct MatrixGroupElem{RE<:RingElem, T<:MatElem{RE}} <: AbstractMatrixGroupElem
+   parent::MatrixGroup{RE, T}
+   elm::T
+   X::GapObj
+
+   function MatrixGroupElem(G::MatrixGroup{RE,MatElem{RE}}, x::MatElem{RE}) where RE <: RingElem
+      z = new{RE, MatElem{RE}}()
+      z.parent = G
+      z.elm = x
+      return z
+   end
+
+   function MatrixGroupElem(G::MatrixGroup{RE,MatElem{RE}}, x::MatElem{RE}, x_gap::GapObj) where RE <: RingElem
+      z = new{RE, MatElem{RE}}()
+      z.parent = G
+      z.elm = x
+      z.X = x_gap
+      return z
+   end
+
+   function MatrixGroupElem(G::MatrixGroup{RE,MatElem{RE}}, x_gap::GapObj) where RE <: RingElem
+      z = new{RE, MatElem{RE}}()
+      z.parent = G
+      z.X = x_gap
+      return z
+   end
+end
+
+ring_elem_type(::Type{MatrixGroup{S,T}}) where {S,T} = S
+mat_elem_type(::Type{MatrixGroup{S,T}}) where {S,T} = T
+
+########################################################################
+#
+# Basic
+#
+########################################################################
+
+function Base.show(io::IO, x::MatrixGroup)
+   if isdefined(x, :descr)
+      if x.descr==:GU || x.descr==:SU
+         print(io, string(x.descr), "(",x.deg,",",characteristic(x.ring)^(div(degree(x.ring),2)),")")
+      else
+         print(io, string(x.descr), "(",x.deg,",",order(x.ring),")")
+      end
+   else
+      print(io, "Matrix group of degree ", x.deg, " over ", x.ring)
+   end
+end
+
+function Base.show(io::IO, x::MatrixGroupElem)          #TODO: is this correct?
+   if isdefined(x, :elm)
+      display(x.elm)
```

Comment:
This seems weird to use `display` here, but at least this expression must use `io`.

Reply:
The difference between `display` and `print(io, ...` is that in the second case the `n x n` entries of a matrix are displayed on a single row. That's why I prefer the first option. But I think this depends on the Oscar version, because I got different results working with different laptops.

Reply:
Maybe you are looking for `show(stdout, "text/plain", x)` ?